90.md

```diff
@@ -67,6 +67,37 @@ All tags are optional.
 * `relays`: List of relays where Service Providers SHOULD publish responses to
 * `p`: Service Providers the customer is interested in. Other SPs MIGHT still choose to process the job
 
+## Encrypted Params 
+
+If the user wants to keep the input parameters a secret, they can encrypt the `i` and `param` tags with the service provider's 'p' tag and add it to the content field. The user also indicates whether the output should be encrypted or not as one of the parameters.
+
+```json
+[
+    [ "i", "what is the capital of France? ", "prompt" ],
+    [ "param", "model", "LLaMA-2" ],
+    [ "param", "max_tokens", "512" ],
+    [ "param", "temperature", "0.5" ],
+    [ "param", "top-k", "50" ],
+    [ "param", "top-p", "0.7" ],
+    [ "param", "frequency_penalty", "1" ],
+    [ "param", "output", "encrypted"]
+]
+
+```
+
+This param data will be encrypted and added to the `content` field and `p` tag should be present 
```

Comment:
How about defining a new input type in addition to the existing ones(`text`, `job`, `event`, `url`) ?

The protocol currently defines the input tag to be`[ "i", <data>, <type>, <optional relay>, <marker> ]`. Perhaps introduce an new type called `encrypted`. i.e.
```
[ "i", <data>, <type>, <optional relay>, <marker> ]
[ "i", BE2Y4xvS6HI..., "encrypted", "04f74530a6ede6b24731b976b8e78fb449ea61f40ff10e3d869a3030c4edc91f" ]

... where in this example
data = the encrypted content
type = encrypted
relay = omitted
marker = SP's pubkey
```

Reply:
I've thoroughly considered this. I don't want to add the service provider's public key again. Additionally, if the parameters are encrypted, only the service provider who can decrypt them can process the request. So, having a `p` tag refers to that specific service provider. Since the `content` field is empty anyway, it's best to store the encrypted data similarly to NIP-51 private lists. Therefore, if both the `p` tag and `content` exist, other service providers can ignore them, while the owner of the 'p' tag's public key can process the request. Start by decrypting the 'content' to view all the tags and then examine other tags to process the request.